RN 13.0.1 to PROD (#1446)
mia-platform/documentation: docs/release-notes/v13.0.1.md

##### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* When a Javascript custom cast function returns a Javascript Date object, now the service properly maps the type into the internal one

For more information check out the service [changelog](/runtime_suite/projection-storer/changelog.md).

## Microfrontend Composer

### Back-Kit Library

The new version `v1.5.3` of the back-kit library is available!

Refer to back-kit [changelog](/microfrontend-composer/back-kit/changelog.md) for finer-grained details of new versions.

#### Bug Fix

This version addresses the following bugs:

- Fixed bug that caused browser not to follow logout redirection
- Fixed broken link in readonly file fields
- Fixed the `bk-export-modal` bug related to the duplication of date format options when changing page
- The `customMessageOnAbsentDate` property of `bk-card` component now supports boolean values

## Marketplace

### New Marketplace Components

### Marketplace Updates

## How to update your Console

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100_how-to-upgrade.md) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ ``.

Reviewer: ```suggestion
For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100_how-to-upgrade.md) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v13.3.0`.

```

### Enable Microservice metric gathering in a `Monitoring` tab

From now on, the Microservice monitoring can be managed in the `Monitoring` tab of each microservice.

### Bug Fix

This version addressed the following bugs:

* fixed a bug that wrongly flashed the Create Project button when entering a Company without the Company Owner role;
* fixed the flicker of nodes and edges on configuration or selection changes in Flow Manager section;
* fixed a bug that caused an error on configuration save when Console version is increased from v12 to v13 and the Flow Manager Configurator is not opened;
* fixed a bug that caused a malfunction in reading environment variables in the 'variables.env' file when creating and first configuring a new environment in console;
* fixed bug in project creation that prevented the user from finding the default template set by Blueprint section;
* fixed bug that caused a problem with redirects in Runtime area;
* fixed bug that prevented saving configuration on projects with Azure as provider, if the corresponding project was empty.

## Fast Data

### Fast Data Configurator

> #### Generate from sample functionality in Single View section
> 
> Added the "generate from sample" functionality also to Single View section allowing users to generate Single Views starting from a JSON or CSV data sample

Reviewer: ```suggestion
#### Generate Single view fields from data sample

From the `Import fields` button, an option has been added that allows you to generate fields ** From a data sample**. 
Doing this will replace the current fields with those contained within the file.
The supported file extensions are: `.csv` and `.json`.
```

### Git repository name validation accepts aldo dots

In the creation of a template or an example, you can now insert dots inside the Git Repository Name, as the validation now allows such inclusion.

### Human readable error validations in Create new branch modal & Project creation page

Error validations when trying to create a new Branch or a new Project appear now more human readable, improving clarity and user understanding.

### Enable Microservice metric gathering in a `Monitoring` tab

From now on, the Microservice monitoring can be managed in the `Monitoring` tab of each microservice.

### Generate CRUD collection fields from data sample

From the `Import fields` button, an option has been added that allows you to generate fields ** From a data sample**. 
Doing this will replace the current fields with those contained within the file.
The supported file extensions are: `.csv` and `.json`.
For more information, visit the [dedicated documentation](/development_suite/api-console/api-design/crud_advanced.md#how-to-generate-from-data-sample-your-crud-fields).

> ### Bug Fix

Reviewer: ```suggestion

### Bug Fix

```

From the `Sidecars` section, you can now clone an existing configuration of a Sidecar to other Microservices. You can perform such action by simply clicking on the new "Add to another microservice" button available in the detail of each Sidecar and select the configuration you want to clone and the target Microservices in which to attach such Sidecar configuration. 

For more information, please refer to the [Sidecars documentation page](/console/design-your-projects/sidecars.md#clone-a-sidecar).

### Improvements

### Improved Console navigation with `ctrl/cmd + k`

The navigation with `ctrl/cmd + k` has been revised and improved for a better user experience when navigating among resources. Users can now easily see which Companies and Projects they belong to and can quickly search within these resources and seamlessly jump to them.

### Interpolation in Annotations and Labels for Marketplace items

When defining a new service on the Markeplace, it is now possible to configure the desired interpolations for Annotations and Labels in it, so that when the respective resource is created on a given Project on Console, the interpolated values are correctly set.

> ### Delete a Microservice in `Advanced` tab

Reviewer: ```suggestion
### Delete a Microservice in Advanced tab

```

The experience of deleting a Microservice has been revised: from now on, if you want to delete a Microservice, you need to do so within the `Advanced` tab in the Microservice detail page.

> ### Raw Manifest Configuration in `Advanced` tab

Reviewer: ```suggestion
### Raw Manifest Configuration in Advanced tab

```

Configuring the raw manifest of a Microservice is now possible within the `Advanced` tab in the Microservice detail page.

### Git repository name validation accepts aldo dots

In the creation of a template or an example, you can now insert dots inside the Git Repository Name, as the validation now allows such inclusion.

### Human readable error validations in Create new branch modal & Project creation page

Error validations when trying to create a new Branch or a new Project appear now more human readable, improving clarity and user understanding.

> ### Enable Microservice metric gathering in a `Monitoring` tab

Reviewer: ```suggestion
### Enable Microservice metric gathering in a Monitoring tab

```

From now on, the Microservice monitoring can be managed in the `Monitoring` tab of each microservice.

### Generate CRUD collection fields from data sample

From the `Import fields` button, an option has been added that allows you to generate fields ** From a data sample**. 
Doing this will replace the current fields with those contained within the file.
The supported file extensions are: `.csv` and `.json`.
For more information, visit the [dedicated documentation](/development_suite/api-console/api-design/crud_advanced.md#how-to-generate-from-data-sample-your-crud-fields).

### Bug Fix

This version addressed the following bugs:

* fixed a bug that wrongly flashed the Create Project button when entering a Company without the Company Owner role;
* fixed the flicker of nodes and edges on configuration or selection changes in Flow Manager section;
* fixed a bug that caused an error on configuration save when Console version is increased from v12 to v13 and the Flow Manager Configurator is not opened;

> * fixed a bug that caused a malfunction in reading environment variables in the 'variables.env' file when creating and first configuring a new environment in console;

Reviewer 2: ```suggestion
* fixed a bug that caused a malfunction in reading environment variables in the '.env' file when creating and first configuring a new environment in console;

```

Reviewer 2: `variables.env` is only true with Kustomize, other configs have the specific environment name, i'd use `.env` that's more generic

##### Improvements

* When the storage component receives an empty list of projection records, instead of treating it as an error now it logs a warning message and it proceeds processing the other topic-partition ingestion events

##### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* When a Javascript custom cast function returns a Javascript Date object, now the service properly maps the type into the internal one

For more information check out the service [changelog](/runtime_suite/projection-storer/changelog.md).

## Microfrontend Composer

### Back-Kit Library

The new version `v1.5.3` of the back-kit library is available!

Refer to back-kit [changelog](/microfrontend-composer/back-kit/changelog.md) for finer-grained details of new versions.

#### Bug Fix

This version addresses the following bugs:

- Fixed bug that caused browser not to follow logout redirection
- Fixed broken link in readonly file fields
- Fixed the `bk-export-modal` bug related to the duplication of date format options when changing page
- The `customMessageOnAbsentDate` property of `bk-card` component now supports boolean values

## How to update your Console

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100_how-to-upgrade.md) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v13.3.2`.

Reviewer: ```suggestion
For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100_how-to-upgrade.md) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v13.3.3`.

```

* fixed a bug that caused a malfunction in reading environment variables in the '.env' file when creating and first configuring a new environment in Console;
* fixed bug in project creation that prevented the user from finding the default template set by Blueprint section;
* fixed bug that caused a problem with redirects in Runtime area;
* fixed bug that prevented saving configuration on projects with Azure as provider, if the corresponding project was empty;
* fixed a bug in the sshUrl management during service creation

## Fast Data

### Fast Data Configurator

#### Generate Single view fields from data sample

From the `Import fields` button, an option has been added that allows you to generate fields ** From a data sample**. 
Doing this will replace the current fields with those contained within the file.
The supported file extensions are: `.csv` and `.json`.
For more information, visit the [dedicated documentation](/fast_data/configuration/single_views.md#generate-single-view-fields-from-data-sample)

#### Bug Fix

This version addressed the following bug:

* Fixed a bug allowing to create cast functions with invalid names (e.g., starting with a capital letter)

> 

Reviewer: ```suggestion
* when the projection storer is attached to a projection that has a field of type boolean to which the default cast function defaultCastStringToBoolean is applied, at configuration saving this cast function was saved with an incorrect name, causing an error when the service starts. 
```

### Raw Manifest Configuration in `Advanced` tab

Configuring the raw manifest of a Microservice is now possible within the `Advanced` tab in the Microservice detail page.

### Git repository name validation accepts aldo dots

In the creation of a template or an example, you can now insert dots inside the Git Repository Name, as the validation now allows such inclusion.

### Human readable error validations in Create new branch modal & Project creation page

Error validations when trying to create a new Branch or a new Project appear now more human readable, improving clarity and user understanding.

### Enable Microservice metric gathering in a `Monitoring` tab

From now on, the Microservice monitoring can be managed in the `Monitoring` tab of each microservice.

### Generate CRUD collection fields from data sample

From the `Import fields` button, an option has been added that allows you to generate fields ** From a data sample**. 
Doing this will replace the current fields with those contained within the file.
The supported file extensions are: `.csv` and `.json`.

> 

Reviewer: ```suggestion
For more information, visit the [dedicated documentation](/development_suite/api-console/api-design/crud_advanced.md#how-to-generate-from-data-sample-your-crud-fields).
```

### Bug Fix

This version addressed the following bugs:

* fixed a bug that wrongly flashed the Create Project button when entering a Company without the Company Owner role;
* fixed the flicker of nodes and edges on configuration or selection changes in Flow Manager section;
* fixed a bug that caused an error on configuration save when Console version is increased from v12 to v13 and the Flow Manager Configurator is not opened;
* fixed a bug that caused a malfunction in reading environment variables in the 'variables.env' file when creating and first configuring a new environment in console;
* fixed bug in project creation that prevented the user from finding the default template set by Blueprint section;
* fixed bug that caused a problem with redirects in Runtime area;
* fixed bug that prevented saving configuration on projects with Azure as provider, if the corresponding project was empty.

## Fast Data

### Fast Data Configurator

#### Generate Single view fields from data sample

From the `Import fields` button, an option has been added that allows you to generate fields ** From a data sample**. 
Doing this will replace the current fields with those contained within the file.

> The supported file extensions are: `.csv` and `.json`.

Reviewer: ```suggestion
The supported file extensions are: `.csv` and `.json`.
For more information, visit the [dedicated documentation](/fast_data/configuration/single_views.md#generate-single-view-fields-from-data-sample)
```

##### Improvements

* When the storage component receives an empty list of projection records, instead of treating it as an error now it logs a warning message and it proceeds processing the other topic-partition ingestion events

##### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* When a Javascript custom cast function returns a Javascript Date object, now the service properly maps the type into the internal one

For more information check out the service [changelog](/runtime_suite/projection-storer/changelog.md).

## Microfrontend Composer

### Back-Kit Library

The new version `v1.5.3` of the back-kit library is available!

Refer to back-kit [changelog](/microfrontend-composer/back-kit/changelog.md) for finer-grained details of new versions.

#### Bug Fix

This version addresses the following bugs:

- Fixed bug that caused browser not to follow logout redirection
- Fixed broken link in readonly file fields
- Fixed the `bk-export-modal` bug related to the duplication of date format options when changing page
- The `customMessageOnAbsentDate` property of `bk-card` component now supports boolean values

> ## Marketplace
> 
> ### New Marketplace Components
> 
> ### Marketplace Updates
> 

Reviewer: ```suggestion
```

---
id: v13.0.1
title: Version 13.0.1 Release Notes
sidebar_label: v13.0.1
image: "img/release-note-link-preview.png"
---

_June 06th, 2024_

:::info
Mia-Platform Console v13.0.1 is **now in Preview** and will be generally available on June 20th.

Console SaaS users can try out v13.0.1 latest improvements in Preview! Open a Service Request to ask for the creation of a sandbox Company in case you do not have access to any Company.

For self-hosted installations, please read the [following guidelines](#how-to-update-your-console).

> :::

Author: should be removed if we go directly in production
